### PR TITLE
PR - Updating website (layout, descriptions, and SRS types)

### PR DESCRIPTION
**Pull request by danscime**
This should be a PR for now as a lot of semi-unrelated items are starting to get mixed together.

This PR updates:
- The style of the website
- Adds some intro and section information
- Adds descriptions to each example
- Cleans up the links for PDF and HTML versions of the examples (#1596)

### CODE REVIEW

**Comment by danscime** [comment]
Not sure why the failure had occurred, merged master as a hopeful fix.

**Comment by Mornix** [comment]
Took a quick look:
`desc` is described as a `maybeField`. And when the site is being built, a `desc` is really `Nothing`. Wrap the call to `desc` in an if-block similar to what is done for `src`.

File: code/website/index.html [hunk 1]
```diff
@@ -1,22 +1,47 @@
 <h2>Drasil - Generate All the Things!</h2>
+<p id="center"><a href="https://github.com/JacquesCarette/Drasil">Link to GitHub repository</a></p>
+<hr>
+<br />
```
Comment (Mornix) [inline code/website/index.html]: Pedantic:
`br` is a [void element](https://www.w3.org/TR/2011/WD-html-markup-20110405/syntax.html#syntax-elements) and thus it is more correct (in my opinion) as `<br>` as opposed to `<br/>`.

File: code/website/index.html [hunk 2]
```diff
@@ -1,22 +1,47 @@
...
+  <a href="#caseStudies">case study artifacts</a>
+  and <a href="#packages">package dependency graphs</a> from the Drasil repository.
+  The case study artifacts include the Software Requirements Specification (SRS) for the case study,
+  which specifies what the program sets out to achieve.
+  The package dependency graphs shows the hierarchy of modules within each package.
+  The footer of this page contains the continuous integration build of the project,
+  as well as the commit number that the build and artifacts are based off of.
+</p>
+<br />
+<a name="caseStudies"></a>
```
Comment (Mornix) [inline code/website/index.html]: `<a name>` is [not valid in HTML5](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a). And an empty link with an "id" is incorrect here. The id `caseStudies` should be given to the `<h3>` on the following line.

File: code/website/index.html [hunk 3]
```diff
@@ -1,22 +1,47 @@
...
-    $-for(srs)$
-      <li><a href="$url$">$name$ SRS [$type$]</a></li>$sep-$
-    $endfor$
+      <li>$name$ SRS
+        $-for(srs)$
+          <a href="$url$">[$type$]</a>
+        $endfor$
+      </li>
     $-if(src)$
-      <li><a href="$src-$">Generated Code</a></li>
```
Comment (Mornix) [inline code/website/index.html]: Any particular reason the `-` was removed here? Does it not affect the generated file's spacing?

File: code/website/index.html [hunk 4]
```diff
@@ -1,22 +1,47 @@
...
     </ul>
+    <br />
   </li>$sep-$
 $endfor$
 </ul>
 $if(docsUrl)-$
 <h4><a href="$docsUrl$">Haskell Documentation</a></h4>
 $-endif$
-Package Dependencies
+<a name="packages"></a>
```
Comment (Mornix) [inline code/website/index.html]: Ditto

File: code/website/site.hs [hunk 5]
```diff
@@ -11,30 +11,52 @@ import System.Directory (doesDirectoryExist, doesFileExist, listDirectory)
...
 getExtensionName _ = error "Expected some extension."
 
+-- returns a list of tuples, where each tuple is (Name, string) where string is "HTML" or "PDF"
 getSRS :: [Name] -> SRSVariants
 getSRS = map (\x -> (x, getExtensionName $ takeExtension x)) . filter (\x -> any (`endswith` x) [".pdf", ".html"])
 
+zip4 :: [a] -> [b] -> [c] -> [d] -> [(a,b,c,d)]
+zip4 (a:as) (b:bs) (c:cs) (d:ds) = (a,b,c,d) : zip4 as bs cs ds
+zip4 _ _ _ _ = []
+
```
Comment (Mornix) [inline code/website/site.hs]: [Builtin to base](https://hackage.haskell.org/package/base-4.7.0.0/docs/Data-List.html#v:zip4)

File: code/website/css/styles.css [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+  color: blue;
+}
+a:hover, a:visited:hover  {
+  background-color: greenyellow;
+}
+a:visited:active {
+  color: purple;
+}
+a:active {
+  opacity: 0.5;
```
[the remark this answers is not in the document]
Comment (danscime) [reply code/website/css/styles.css]: Just trying out different effects, can easily be changed.

File: code/website/index.html (hunk 3 above)
Comment (danscime) [reply code/website/index.html]: Does not seem to make a difference on my browser.

File: code/website/index.html (hunk 1 above)
Comment (danscime) [reply code/website/index.html]: I can change them `<br>`.

File: code/website/site.hs (hunk 5 above)
Comment (danscime) [reply code/website/site.hs]: Will remove and use built in version.

File: code/scripts/deploy_stage.sh [hunk 7]
```diff
@@ -55,6 +55,12 @@ copy_graphs() {
   cp -r "$CUR_DIR$GRAPH_FOLDER". "$GRAPH_FOLDER"
 }
 
+copy_descs() {
+  rm -r descriptions >/dev/null 2>&1  # Printing an error message that a directory doesn't exist isn't the most useful.
+  mkdir -p descriptions
+  cp -r "$CUR_DIR"website/descriptions/. descriptions/
+}
+
```
[the remark this answers is not in the document]
Comment (danscime) [reply code/scripts/deploy_stage.sh]: I am pretty sure the the files were directly pulled. I think initially I copied the script to make it similar to `src`. It's been fixed now.

**Comment by danscime** [comment]
All good and ready to merge.

**Comment by JacquesCarette** [comment]
@Mornix you wanted me to hold off on this - please tell me when this can be merged in.

**Comment by danscime** [comment]
@Mornix and I worked through the current "bugs".
I believe we are good to merge now.
Newer changes will be done on a separate branch.

**Comment by Mornix** [comment]
One last [small correction](https://travis-ci.org/JacquesCarette/Drasil/builds/550438446#L4382) and this is good to merge.

**Comment by danscime** [comment]
The @Mornix the correction was to put the link within h3 correct? Or was there something else?

**Comment by danscime** [comment]
Added one more update based on [this line](https://travis-ci.org/JacquesCarette/Drasil/builds/550438446#L4236)

**Pull request merged**